### ISSUE TITLE
Inconsistent behaviour of ?-operator in shader language on Mac

### ISSUE THREAD

**Issue by peterhoglund** [godotengine/godot#46492]
**Godot version:**
3.2.3 Stable Official

**OS/device including version:**
MacBook Pro – MacOS Big Sur v. 11.2.1
Radeon Pro 560 4 GB
Intel HD Graphics 630 1536 MB

Windows 10

**Issue description:**
There are inconsistencies in how Godot's shader language handles the ?-operator between Mac and Windows, and in different contexts on Mac.

**Steps to reproduce:**
```
shader_type canvas_item;

uniform bool text_check;
uniform bool uv_check;

void fragment()
{
	vec4 text = texture(text_check ? SCREEN_TEXTURE : TEXTURE, uv_check ? UV : SCREEN_UV);
	
	COLOR = text;
}
```

Consider this code. On Mac, checking and unchecking `uv_check` and `text_check` will flip between `UV` and `SCREEN_UV` as you would expect, but not between `SCREEN_TEXTURE` and `TEXTURE`. It will only default to the last option, `TEXTURE` in this case.

Both are allowed, there is no error message, and `text_check` is working on Windows, which makes me suspect this is a bug on Mac and not intentional. 

**Minimal reproduction project:**

**Comment by Calinou** [comment]
The bug likely resides in the graphics driver/shader compiler rather than in the operating system. I'm not sure if we can work around it.

**Comment by clayjohn** [comment]
Yeah, Godot doesn't interfere with the ``?`` operator. This is clearly a driver/hardware issue. 

As a workaround try using proper ``if/else`` statements for your conditions. While the syntax is a little more verbose, it will likely make your GLSL compiler happy.
